x-pack/plugins/observability_solution/dataset_quality/server/routes/data_streams/routes.ts

```diff
@@ -123,6 +125,32 @@ const nonAggregatableDatasetsRoute = createDatasetQualityServerRoute({
   },
 });
 
+const degradedFieldsRoute = createDatasetQualityServerRoute({
+  endpoint: 'GET /internal/dataset_quality/data_streams/{dataStream}/degraded_fields',
+  params: t.type({
+    path: t.type({
+      dataStream: t.string,
+    }),
+    query: rangeRt,
+  }),
+  options: {
+    tags: [],
+  },
+  async handler(resources): Promise<DegradedField[]> {
+    const { context, params } = resources;
+    const { dataStream } = params.path;
+    const coreContext = await context.core;
+
+    const esClient = coreContext.elasticsearch.client.asCurrentUser;
+
+    return await getDegradedFields({
```

Comment:
**nit**: not sure if this is handled somewhere by `createDatasetQualityServerRoute`, but would be great to use the RT response type to encode the delivered response as security check in case something unexpected comes from the ES query.

Reply:
The client side code does the response validation with the RT

https://github.com/achyutjhunjhunwala/kibana/blob/07c0fc66e337cc3e102d0012220727dc734b3302/x-pack/plugins/observability_solution/dataset_quality/public/services/data_stream_details/data_stream_details_client.ts#L74-L99